lib/CSSStyleDeclaration.js

```diff
@@ -162,133 +99,78 @@ class CSSStyleDeclaration {
-            continue;
-          }
-          if (item.type === "Rule") {
-            shouldSkipNext = true;
-            continue;
-          }
-          if (shouldSkipNext === true) {
-            shouldSkipNext = false;
-            continue;
-          }
-          const {
-            important,
-            property,
-            value: { value }
-          } = item;
-          if (typeof property === "string" && typeof value === "string") {
-            const priority = important ? "important" : "";
-            const isCustomProperty = property.startsWith("--");
-            if (isCustomProperty || hasVarFunc(value)) {
-              if (properties.has(property)) {
-                const { priority: itemPriority } = properties.get(property);
-                if (!itemPriority) {
-                  properties.set(property, { property, value, priority });
-                }
-              } else {
-                properties.set(property, { property, value, priority });
-              }
-            } else {
-              const parsedValue = parsePropertyValue(property, value, {
-                globalObject: this._global
-              });
-              if (parsedValue) {
-                if (properties.has(property)) {
-                  const { priority: itemPriority } = properties.get(property);
-                  if (!itemPriority) {
-                    properties.set(property, { property, value, priority });
-                  }
-                } else {
-                  properties.set(property, { property, value, priority });
-                }
-              } else {
-                this.removeProperty(property);
-              }
-            }
-          }
-        }
-        const parsedProperties = prepareProperties(properties, {
-          globalObject: this._global
-        });
-        for (const [property, item] of parsedProperties) {
-          const { priority, value } = item;
-          this._priorities.set(property, priority);
-          this.setProperty(property, value, priority);
-        }
+        const properties = this._extractDeclarations(valueObj.children);
+        this._applyProperties(properties);
       }
     } catch {
       return;
     }
-    this._setInProgress = false;
-    if (typeof this._onChange === "function") {
-      this._onChange(this.cssText);
-    }
+    this._updating = false;
+    this._notifyChange();
   }
 
+  /**
+   * The number of properties in the declaration block.
+   *
+   * @type {number}
+   */
   get length() {
     return this._length;
   }
 
-  // This deletes indices if the new length is less then the current length.
-  // If the new length is more, it does nothing, the new indices will be
-  // undefined until set.
+  /**
+   * This deletes indices if the new length is less then the current length.
+   * If the new length is more, it does nothing, the new indices will be undefined until set.
+   *
+   * @param {number} len
+   */
   set length(len) {
```

Comment:
This setter isn't necessary for jsdom, nor is it in the spec. Can we delete it? Or are there internal uses of some sort? Probably those could be replaced...

Reply:
I guess this and the index() might need to be cleaned up in a follow-up. Overall it would be nicer if we did not include the indexed properties on the public API, and let the webidl2js layer do them instead.